rio_cogeo/scripts/cli.py

```diff
@@ -258,24 +262,33 @@ def create(
         output_profile["blockxsize"] = blocksize
         output_profile["blockysize"] = blocksize
 
-    if web_optimized:
-        overview_blocksize = blocksize or 512
-
     config.update(
         {
             "GDAL_NUM_THREADS": threads,
             "GDAL_TIFF_INTERNAL_MASK": os.environ.get("GDAL_TIFF_INTERNAL_MASK", True),
-            "GDAL_TIFF_OVR_BLOCKSIZE": str(overview_blocksize),
         }
     )
 
+    tilematrixset: TileMatrixSet | None
     if tms:
         with open(tms, "r") as f:
             tilematrixset = TileMatrixSet(**json.load(f))
-
+    elif web_optimized:
+        tilematrixset = morecantile.tms.get("WebMercatorQuad")
     else:
         tilematrixset = None
 
+    gdal_tif_ovr_blocksize: int | None
+    if overview_blocksize:
+        gdal_tif_ovr_blocksize = overview_blocksize
+    elif blocksize:
+        gdal_tif_ovr_blocksize = blocksize
+    elif output_profile.get("blockxsize", -1) == output_profile.get("blockysize", -2):
+        gdal_tif_ovr_blocksize = blocksize
+
+    if gdal_tif_ovr_blocksize:
+        config.update({"GDAL_TIFF_OVR_BLOCKSIZE": gdal_tif_ovr_blocksize})
```

Comment:
why do we need this?

Reply:
I tried to keep the previous behaviour, but instead of relying on web-optimization for it, I changed it to be determined by user specified blocksize(s). But this one is part of the problem I describe in the last bullet point of the PR description.

But the main problem I have here is that if I [understand GDAL documentation](https://gdal.org/drivers/raster/gtiff.html#overviews), it should match tile matrix set specification. But with the current way of processing the command argument, I think this is not the case.

I tried to set it as well as possible to match current test cases, but I think that there is something unclear there.

Note that if you prefer, I can change back the code as it was (i.e. force overview blocksize to input blocksize).
In such case, maybe I can raise a different issue focused on this subject, and keep the current PR only focused on web_optimized in python API.

Reply:
agreed! 

we should set the block size based on https://github.com/developmentseed/morecantile/blob/main/morecantile/models.py#L282-L297 I think for now we should try to match what was in place and we can change it in another PR

Reply:
FYI: I remember why we had this

by default we have the overview block size set at 128 or `GDAL_TIFF_OVR_BLOCKSIZE`. When we were using the `web-optimized` option we basically wanted the overview block size to be the same as the high resolution blocksize (hence why we have `overview_blocksize = blocksize or 512`). 

The default profiles use `512` block size which explain why we have `blocksize or 512`. 

### Next

This is a bit tricky but when we want to web-optimized a file we want all the blocksize to be the same. Right now we default to the profile block size but we should default to the TMS blocksize

Reply:
From my understanding, there might be two actions that could clarify/improve configuration: 

 1. Some cli parameters conflict with each other. For example, if user set `--overview-blocksize=256`, `--blocksize=512` and `--web-optimized`, cogeo receives conflicting configuration. I think that some validation and warnings could be emitted in such cases.
2. Identified conflicting parameters involve target tile matrix set definition. One way to detect such conflict early and simplify configuration would be to always:
     a. build a TileMatrixSet from user input parameters (for example, if user only set `blocksize`, a TileMatrixSet with the same CRS and bbox as input dataset could be built). 
    b. Then, and only then, prepare copy parameters from the tile matrix set. 

That would surely involve more work internally, but it would cleanly split concerns: user input validation/interpretation one one side, and gdal copy parameterization on the other side. And the bridge between them would be the TileMatrixSet object (at least for tiling and level of details). 
Plus, we would then always have a TileMatrixSet specification as intermediate representation, which, I think, would ease debug and configuration sharing.